go/beacon/api/scrape.go

```diff
@@ -0,0 +1,144 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/scrape"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	"github.com/oasisprotocol/oasis-core/go/common/pubsub"
+	"github.com/oasisprotocol/oasis-core/go/consensus/api/transaction"
+)
+
+var (
+	// MethodSCRAPECommit is the method name for a SCRAPE commitment.
+	MethodSCRAPECommit = transaction.NewMethodName(ModuleName, "SCRAPECommit", SCRAPECommit{})
+
+	// MethodSCRAPEReveal is the method name for a SCRAPE reveal.
+	MethodSCRAPEReveal = transaction.NewMethodName(ModuleName, "SCRAPEReveal", SCRAPEReveal{})
+)
+
+// SCRAPEParameters are the beacon parameters for the SCRAPE backend.
+type SCRAPEParameters struct {
+	Participants  uint64 `json:"participants"`
+	Threshold     uint64 `json:"threshold"`
+	PVSSThreshold uint64 `json:"pvss_threshold"`
+
+	CommitInterval  int64 `json:"commit_interval"`
+	RevealInterval  int64 `json:"reveal_interval"`
+	TransitionDelay int64 `json:"transition_delay"`
+
+	GasCosts transaction.Costs `json:"gas_costs,omitempty"`
+}
+
+const (
+	// GasOpSCRAPECommit is the gas operation identifier for a commit.
+	GasOpSCRAPECommit transaction.Op = "scrape_commit"
+	// GasOpSCRAPEReveal is the gas operation identifier for a reveal.
+	GasOpSCRAPEReveal transaction.Op = "scrape_reveal"
+)
+
+// DefaultGasCosts are the "default" gas costs for operations.
+var DefaultGasCosts = transaction.Costs{
+	GasOpSCRAPECommit: 1000,
+	GasOpSCRAPEReveal: 1000,
```

Comment:
Per discussion I'm just going to remove the gas later, once the reactor is fixed (the Tx-es are functionally free anyway because the validator submits their own commit/reveal).